### PR TITLE
Refactored codebase to share primitives between front end and back end

### PR DESCRIPTION
**Pull request by lucaswoj**
Depends on #124 

Logically merged frontend and backend value systems.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -4,6 +4,9 @@ apply plugin: 'jacoco'
 apply plugin: 'coveralls'
 apply plugin: 'checkstyle'
 
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
+
```
Comment (lucaswoj) [inline build.gradle:10]: This change is unrelated but tiny. It enforces the use of Java 8 from Gradle.

File: checkstyle.xml [hunk 2]
```diff
@@ -48,16 +48,6 @@ page at http://checkstyle.sourceforge.net/config.html -->
...
-    <module name="ImportOrder">
-      <!-- Checks for out of order import statements. -->
-
-      <property name="severity" value="warning"/>
-      <property name="groups" value="com.google,android,junit,net,org,java,javax"/>
-      <!-- This ensures that static imports go first. -->
-      <property name="option" value="top"/>
-      <property name="tokens" value="STATIC_IMPORT, IMPORT"/>
-    </module>
-
```
Comment (lucaswoj) [inline checkstyle.xml:64]: This is also unrelated. It was just bothering me.

**Comment by lucaswoj** [comment]
Git diffs make this look major but most of the changes in here are refactoring and superficial changes to classes to fit the right interface. There is no new logic in here :smile:

**Comment by lucaswoj** [comment]
I've completely unified frontend and backend types. This is the first step to implementing parsing and elaboration.

**Comment by mtrberzi** [comment]
Very nice. This is what I was waiting to see before making a second attempt at elaboration. I will have more time to look at this later on...in the meantime, want to resolve conflicts and push again?

**Comment by lucaswoj** [comment]
Done!

**Comment by mtrberzi** [comment]
NICE! I really like the way this is organized. (I didn't have to look at any of the code to figure out what was going on, just imported the project and looked at the file structure.) This is going to be so much easier to work with in both the frontend and backend. :shipit:
